### PR TITLE
add goleak

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <i.can.feel.gravity@gmail.com>

### Description:
add goleak

### Environment:

- Golang Version: 1.14
- Docker Version: 19.03.5
- Kubernetes Version: 1.17.3
- NGT Version: 1.9.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/net/http/middleware/timeout_test.go [hunk 1]
```diff
@@ -16,6 +16,7 @@
 package middleware
 
 import (
+	"errors"
```
Comment (vankichi) [inline internal/net/http/middleware/timeout_test.go:19]: I think `errors` package is not needed.

File: go.mod [hunk 2]
```diff
@@ -45,6 +45,7 @@ require (
 	github.com/kpango/glg v1.5.1
 	github.com/lucasb-eyer/go-colorful v1.0.3
 	github.com/pierrec/lz4/v3 v3.3.2
+	github.com/pkg/errors v0.8.1
```
Comment (vankichi) [inline go.mod:48]: I could not understand why this package is added into `go.mod`...

Comment (kpango) [reply go.mod:48]: i don't know

Comment (kevindiu) [reply go.mod]: I think it is caused by the `defaultCheckFunc`
```golang
	defaultCheckFunc := func(w want, got [][]float32) error {
		if !reflect.DeepEqual(got, w.want) {
			return errors.Errorf("got = %v, want %v", got, w.want)
		}
		return nil
	}
```

Comment (vankichi) [reply go.mod]: I did not realize this. thx

Comment (kevindiu) [reply go.mod]: @vankichi no, I think I made an mistake... seems it is not caused by the `checkfunc`...

Comment (kevindiu) [reply go.mod]: but it is no longer exists now, so it should be ok~

**Comment by vankichi** [comment]
close #351

**Pull request merged**